File: leetcode/easy/3110-score-of-a-string/3110-score-of-a-string.py
# // 3110-score-of-a-string
# // URL: https://leetcode.com/problems/score-of-a-string/description/

class Solution:
    def getAscii(self, c: str) -> int:
        return ord(c)

    def scoreOfString(self, s: str) -> int:
        score = 0
        lastAscii = self.getAscii(s[0])

        for c in s[1:]:
            asciiValue = self.getAscii(c)
            score += abs(asciiValue - lastAscii)
            lastAscii = asciiValue

        return score


print(Solution().scoreOfString("hello"))
# 13

File: leetcode/easy/3110-score-of-a-string/3110-score-of-a-string.ts
// 3110-score-of-a-string
// URL: https://leetcode.com/problems/score-of-a-string/description/
function scoreOfString(s: string): number {
    let lastAscii = 0;
    let score = 0;
  
    lastAscii = s.charCodeAt(0);
  
    for (let i = 1; i < s.length; i++) {
      let ascii = s.charCodeAt(i);
      score += Math.abs(ascii - lastAscii);
      lastAscii = ascii;
    }
  
    return score;
  };
  